Activities/Cryptography/UiPath.Cryptography.Activities/KeyedHashText.cs

```diff
@@ -80,22 +86,27 @@ protected override string Execute(CodeActivityContext context)
             {
                 string input = Input.Get(context);
                 string key = Key.Get(context);
+                SecureString keySecureString = KeySecureString.Get(context);
                 Encoding encoding = Encoding.Get(context);
 
                 if (string.IsNullOrWhiteSpace(input))
                 {
                     throw new ArgumentNullException(Resources.InputStringDisplayName);
                 }
-                if (string.IsNullOrWhiteSpace(key))
+                if (string.IsNullOrWhiteSpace(key) && keySecureString == null)
+                {
+                    throw new ArgumentNullException(Resources.KeyAndSecureStringNull);
+                }
+                if (key != null && keySecureString != null)
                 {
-                    throw new ArgumentNullException(Resources.Key);
+                    throw new ArgumentNullException(Resources.KeyAndSecureStringNotNull);
                 }
                 if (encoding == null)
                 {
                     throw new ArgumentNullException(Resources.Encoding);
                 }
 
-                byte[] hashed = CryptographyHelper.HashDataWithKey(Algorithm, encoding.GetBytes(input), encoding.GetBytes(key));
+                byte[] hashed = CryptographyHelper.HashDataWithKey(Algorithm, encoding.GetBytes(input), key != null ? encoding.GetBytes(key) : encoding.GetBytes(new NetworkCredential("", keySecureString).Password));
```

Comment:
This part

```
key != null ? encoding.GetBytes(key) : encoding.GetBytes(new NetworkCredential("", keySecureString).Password)
```

Is reused in a lot of places in this PR. Can we extract it to a helper method please?

Validations can stay as they are, in theory different activities can have different validation rules, and the checks themselves are simple enough to leave duplicated, but this is a good candidate for a helper method.

Reply:
Extracted as helper method into CryptographyHelper.